pkg/zsio/detector/reader.go

```diff
@@ -0,0 +1,43 @@
+package detector
+
+import (
+	"errors"
+	"io"
+
+	"github.com/mccanne/zq/pkg/zsio/ndjson"
+	"github.com/mccanne/zq/pkg/zsio/raw"
+	"github.com/mccanne/zq/pkg/zsio/zjson"
+	zsonio "github.com/mccanne/zq/pkg/zsio/zson"
+	"github.com/mccanne/zq/pkg/zson"
+	"github.com/mccanne/zq/pkg/zson/resolver"
+)
+
+var ErrUnknown = errors.New("input format not recognized")
+
+func NewReader(r io.Reader, n int, t *resolver.Table) (zson.Reader, error) {
+	recorder := NewRecorder(r)
+	track := NewTrack(recorder)
+	if match(zsonio.NewReader(track, resolver.NewTable())) {
```

Comment:
could zsio and raw be in different places? just to understand the ordering...

Reply:
Yeah, the only dependency should be zjson before ndjson